ArduCopter/GCS_Mavlink.cpp

```diff
@@ -974,6 +974,30 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
         GCS_MAVLINK_Copter::convert_COMMAND_LONG_to_COMMAND_INT(packet, packet_int);
         return handle_command_pause_continue(packet_int);
     }
+
+    case MAV_CMD_DO_CHANGE_ALTITUDE: {
+        // param1 : Altitude in meters
+        // param2 : Frame of new altitude MAV_FRAME   
+        if (!copter.flightmode->in_guided_mode()) {
+            return MAV_RESULT_FAILED;
+        }
+
+        uint8_t frame = packet.param2;
+        // check for supported coordinate frames
+        if (frame != MAV_FRAME_GLOBAL_RELATIVE_ALT && frame != MAV_FRAME_GLOBAL_RELATIVE_ALT_INT) {
```

Comment:
Why just relative frames?

Reply:
Hi, this is just to get a basic implementation going. I think its important to also add other frames I just haven't gone around to handling those yet.

Reply:
We have a convenience function for turning a mavlink frame into a Location-object frame which would make supporting all frames easy.